content/en/docs/concepts/services-networking/dns-pod-service.md

```diff
@@ -157,6 +157,18 @@ pointing to the Pod's IP address. Also, Pod needs to become ready in order to ha
 record unless `publishNotReadyAddresses=True` is set on the Service.
 {{< /note >}}
 
+### Pod's setHostnameAsFQDN field {#pod-sethostnameasfqdn-field}
+
+When a Pod is configured to have fully qualified domain name (FQDN), its hostname is the short hostname. For example, if you have a Pod with the fully qualified domain name `busybox-1.default-subdomain.my-namespace.svc.cluster-domain.example`, then by default the `hostname` command inside that Pod returns `busybox-1` and  the `hostname --fqdn` command returns the FQDN.
+
+When you set `setHostnameAsFQDN: true` in the Pod spec, the kubelet writes the Pod's FQDN into the hostname for that Pod's namespace. In this case, both `hostname` and `hostname --fqdn` return the Pod's FQDN.
+
+{{< note >}}
+In Linux, the hostname field of the kernel (the `nodename` field of `struct utsname`) is limited to 64 characters.
+
+If a Pod enables this feature and its FQDN is longer than 64 character, it will fail to start. The Pod will remain in `Pending` status (`ContainerCreating` as seen by `kubectl`) generating error events regarding its FQDN length. One way of improving user experience for this scenario is to create an [admission webhook controller](/docs/reference/access-authn-authz/extensible-admission-controllers/#admission-webhooks) to control FQDN size when users create top level objects, for example, Deployment.
```

Comment:
Is this page relevant:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names

Reply:
I am not sure we should link to that page, it might be confusing for users. That page is mentioning that some k8s objects follow the same regex as DNS for naming. In this page we are discussing networking concepts, which are not really k8s objects.

Reply:
Right. I guess I found this phrase confusing:
`If a Pod enables this feature and its FQDN is longer than 64 character,`
I'll see if the KEPs add clarification.